### ISSUE TITLE
Apollo federation causes a data graph is missing a valid config error

### ISSUE THREAD

**Issue by LeOndaz** [mirumee/saleor#6439]
Well, I'm trying to use apollo federation with saleor to extend the schema.

steps to reproduce this: create a server like this, I've a plugin that listens for graphql, (ignore, useless for the purpose of this)
```
import {ApolloServer} from 'apollo-server'
import {ApolloGateway} from '@apollo/gateway'

const SALEOR_SERVICE: string = 'http://localhost:8000/graphql/';
const REVIEWS_SERVICE: string = 'http://localhost:8000/plugins/reviews.plugin/';


const gateway = new ApolloGateway({
    serviceList: [
        { name: 'Saleor', url: SALEOR_SERVICE },
        { name: 'Reviews Plugin', url: REVIEWS_SERVICE },
    ],
});

const server = new ApolloServer({
    gateway,
    subscriptions: false
});

server.listen(8001).then(({ url }) => {
  console.log(`🚀 Server ready at ${url}`);
});
```
start your server 
```
npm start
```
The error is 
```
� Server ready at http://localhost:8001/
Encountered error when loading Saleor at http://localhost:8000/graphql/: Syntax Error: Unexpected Name "ObjectWithMetadata"
This data graph is missing a valid configuration. Type Query must define one or more fields.
```
Even though the saleor server /graphql/ itself is running perfectly, this happens when it's used as a federation service. I think the schema is not 100% federation valid, should be easily solvable.

**Comment by maarcingebala** [comment]
Thanks for reporting @LeOndaz, would you be able to send us a PR to fix it since you have Saleor up and running with federation? If not, we'll fix it but I'm afraid not too soon, the team is pretty busy with multichannel now. I'll add it to our backlog anyway.

**Comment by LeOndaz** [comment]
I didn't get it working and I used another workaround, but I've debugged the problem for hours and I've got everything we need to start solving it.
By querying the 
```
{
  _service {
      sdl
   }
{
```
You can find that all ampersands (&) are replaced with commas (old version of graphql)
1) A wide guess, graphene federation repo is unmaintained for 1.5 years and probably it's the one causing this. this is what made me read the source code of the library to see if it's causing any errors, maybe the `get_sdl` function returns a wrong sdl, actually the library is working like a charm, it does what it's supposed to do, the schema is received in graphene federation with commas and this implies that something external made this.
2) turns out that graphene v2 (graphql core) doesn't use ampersands from the beginning and we must use the beta v3 of (graphene, graphene-django) in order to solve this, see [here](https://github.com/graphql-python/graphql-core-legacy/issues/220)
3) I've tested all versions above 3 (beta ones), you need to have 3.x from both of graphene, graphene-django and I did, every combination has a different bug to be fixed (easy ones, but bugs), some of them had a bug of having a mutable default argument in a dataclass related to `account_errors`, other ones had a bug related to a custom JSONField that saleor has.
4) A stupid solution is replacing changing all commas to ampersands in graphene-federation `get_sdl` (workaround, should work)

**Comment by LeOndaz** [comment]
So now the next move is bump graphene/ graphene-django to v3, but this requires some work as there're many things that have changed, I'll see what I can do if this stays unsolved till I finish my current project, a week or two maybe

**Comment by JannikZed** [comment]
hey @LeOndaz - I'm facing the same issue right now, but I don't know exactly why and how. I think some other projects are already using saleor with Graphql federation, at least I've seen some using Graphql queries and mutations that are not originally integrated into the core - and I don't know, how else they could have achieved that.

**Comment by LeOndaz** [comment]
> hey @LeOndaz - I'm facing the same issue right now, but I don't know exactly why and how. I think some other projects are already using saleor with Graphql federation, at least I've seen some using Graphql queries and mutations that are not originally integrated into the core - and I don't know, how else they could have achieved that.

My project is actually using a plugin that adds a new graphql endpoint, not integrated with federation because of this, I know the reasons as mentioned above

**Comment by benoj** [comment]
Hey All - Just running into the same issue as the OP - any movement on this?

**Comment by LeOndaz** [comment]
> Hey All - Just running into the same issue as the OP - any movement on this?

Unfortunately it's still not fixed

**Comment by fowczarek** [comment]
Resolved #7167
